vllm/core/block/cpu_gpu_block_allocator.py

```diff
@@ -178,6 +178,25 @@ def get_num_free_blocks(self, device: Device) -> int:
         """
         return self._allocators[device].get_num_free_blocks()
 
+    def swap(self, blocks: List[Block], source_device: Device,
```

Comment:
docstring pls

also can we add some small comments? the goal is for the code to be immediately readable, e.g. I can know what the intention is behind the `swap_mapping` line without parsing the code

Reply:
also does this currently require that we call `get_and_reset_swaps` every time?

Reply:
That's a problem, needs to move this up to scheduler layer and `get_and_reset_swaps` once every `_schedule()`.

Reply:
Yeah, I think we can aggregate them with a Dict[int, List[int]] for src: many_dst or something. might need to be the inverse, haven't thought through it